scripts/package-lib

```diff
@@ -156,6 +156,29 @@ rsecurl() {
     rcurl --proto '=https' "$@"
 }
 
+# Join elements with the given delimiter
+#
+# Arguments:
+#
+# $1 - Separator string
+# $2... - Elements to join
+#
+# Output:
+#
+# Joined string.
+join-elements() {
+    local sep="$1"
+    local cursep=""
```

Comment:
i don't see how $2 relates to cursep

Reply:
$2 and following arguments are the elements to be joined together.
$cursep is null on the first loop iteration, then is set to the separator in $1 (this is so that we don’t have a trailing separator on the left of the resulting string).

Reply:
that makes sense